quinn/src/endpoint.rs

```diff
@@ -101,10 +101,10 @@ impl Endpoint {
     pub fn new_with_abstract_socket(
         config: EndpointConfig,
         server_config: Option<ServerConfig>,
-        socket: impl AsyncUdpSocket,
+        socket: Box<dyn AsyncUdpSocket>,
         runtime: Arc<dyn Runtime>,
     ) -> io::Result<Self> {
-        Self::new_with_runtime(config, server_config, Box::new(socket), runtime)
+        Self::new_with_runtime(config, server_config, socket, runtime)
```

Comment:
It looks like this signature has converged to being identical with `new_with_runtime`. Want to inline that and update its callsites to use `new_with_abstract_socket` directly for simplicity?

Reply:
Good idea. Removed `new_with_runtime` and replaced it with `new_with_abstract_socket`.